pkg/appgw/internaltypes.go

```diff
@@ -220,3 +224,33 @@ func formatHostname(hostName string) string {
 	// Hostname is NOT empty - prefix it with a dash
 	return fmt.Sprintf("%s-", hostName)
 }
+
+func (listenerID *listenerIdentifier) setHostNames(hostnames []string) int {
```

Comment:
I don't think you are using the `int` returned from the function anywhere.

Also, we can refactor this like
```go
func (listenerID *listenerIdentifier) setHostNames(hostnames []string) {
	listenerID.HostName = hostnames[0]
	hostNameCount:= int(math.Min(float64(len(hostnames)), float64(MaxAllowedHostnames)))
    listenerID.HostNames = hostnames[0:hostNameCount]
}
```

Reply:
We use it to log it in generateListenerID. Do you want me to remove that ?